File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        
        int row = matrix.size();
        int col = matrix[0].size();
        
        unordered_set<int> s1;
        unordered_set<int> s2;
        
        for(int i=0;i<row;i++){
            for(int j=0;j<col;j++){
                if(matrix[i][j] == 0){
                    s1.insert(i);
                    s2.insert(j);
                }
            }
        }
        
        for(auto x : s1){
            for(int i=0;i<col;i++){
                matrix[x][i] = 0;
            }
        }
        
        for(auto x : s2){
            for(int i=0;i<row;i++){
                matrix[i][x] = 0;
            }
        }
    }
};

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        Set<Integer> s1 = new HashSet<>();
        Set<Integer> s2 = new HashSet<>();
        
        for(int i=0;i<matrix.length;i++){
            for(int j=0;j<matrix[0].length;j++){
                if(matrix[i][j] == 0){
                    s1.add(i);
                    s2.add(j);
                }
            }
        }
        
        for(int row : s1){
            for(int i=0;i<matrix[0].length;i++){
                matrix[row][i] = 0;
            }
        }
        
        for(int col : s2){
            for(int i=0;i<matrix.length;i++){
                matrix[i][col] = 0;
            }
        }
        
        
    }
}